azure-pipelines.yml

```diff
@@ -0,0 +1,40 @@
+# Starter pipeline
+# Start with a minimal pipeline that you can customize to build and deploy your code.
+# Add steps that build, run tests, deploy, and more:
+# https://aka.ms/yaml
+
+trigger:
+- develop
+- feature/*
+
+pr:
+- develop
+- feature/*
+
+pool:
+  vmImage: 'ubuntu-latest'
+
+variables:
+  zig_version: 0.4.0
+
+steps:
+- script: |
+    export PYTHONIOENCODING=utf8
+    wget $(curl -s 'https://ziglang.org/download/index.json' | python3 -c "import sys, json; print(json.load(sys.stdin)['$(zig_version)']['x86_64-linux']['tarball'])")
+    tar -xvf zig*
+  displayName: 'Download zig'
+
+- script: zig*/zig build
+  displayName: 'Build kernel'
+
+# Uncomment once mock testing is finished
+#- script: zig*/zig build test
+#  displayName: 'Mocked tests'
+
+- script: |
+    sudo apt-get update
+    sudo apt-get install qemu qemu-system --fix-missing
+  displayName: 'Download qemu'
+  
+- script: zig*/zig build test -Drt-test=true -Dzig-path=zig*/zig
```

Comment:
This might be a big ask for this feature/issue/branch but could you add to your python script that will check for the zig tests being all 'OK' as well as the existing run time tests. So will run the zig tests and check each line is 'OK' and then run the run time tests

Reply:
I'm not sure if checking that the source file tests are all "OK" is necessary as Zig will exit with a non-zero code if any of them fail, causing the Azure step to fail. The same happens for the runtime tests.